src/libs/OptimisticReportNames.ts

```diff
@@ -278,21 +305,56 @@ function computeReportNameIfNeeded(report: Report | undefined, incomingUpdate: O
  * Update optimistic report names based on incoming updates
  * This is the main middleware function that processes optimistic data
  */
-function updateOptimisticReportNamesFromUpdates(updates: OnyxUpdate[], context: UpdateContext): OnyxUpdate[] {
+function updateOptimisticReportNamesFromUpdates(optimisticData: OnyxUpdate[], context: UpdateContext, successData?: OnyxUpdate[], failureData?: OnyxUpdate[]): ProcessedUpdates {
     const {betas, allReports, betaConfiguration} = context;
 
     // Check if the feature is enabled
     if (!Permissions.isBetaEnabled(CONST.BETAS.CUSTOM_REPORT_NAMES, betas, betaConfiguration)) {
-        return updates;
+        return {
+            optimisticData,
+            successData,
+            failureData,
+        };
     }
 
     Log.info('[OptimisticReportNames] Processing optimistic updates for report names', false, {
-        updatesCount: updates.length,
+        updatesCount: optimisticData.length,
     });
 
-    const additionalUpdates: OnyxUpdate[] = [];
+    const additionalOptimisticUpdates: OnyxUpdate[] = [];
+    const additionalSuccessUpdates: OnyxUpdate[] = [];
+    const additionalFailureUpdates: OnyxUpdate[] = [];
+
+    /**
+     * Helper function to add report name updates to the appropriate arrays
+     */
+    function addUpdates(reportID: string, reportNameUpdate: ReportNameUpdate) {
```

Comment:
Let's move this function out of `updateOptimisticReportNamesFromUpdates` instead of nesting it to make this easier to read.